lib/api/core/entrypoints/embedded/protocols/http.js

```diff
@@ -287,9 +277,143 @@ class HttpProtocol extends Protocol {
       data = JSON.stringify(result.content, undefined, indent);
     }
 
-    return data;
+    return Buffer.from(data);
+  }
+
+  /**
+   * Returns a new Readable Stream configured with
+   * uncompression algorithms if needed
+   *
+   * @param  {http.IncomingMessage} request
+   * @return {Array.<stream.Readable>}
+   * @throws {BadRequestError} If invalid compression algorithm is set
+   *                           or if the value does not comply to the
+   *                           way the Kuzzle server is configured
+   */
+  _uncompress(request) {
+    const pipes = [];
+
+    if (request.headers['content-encoding']) {
+      const encodings = request.headers['content-encoding']
+        .split(',')
+        .map(e => e.trim().toLowerCase());
+
+      // encodings are listed in the same order they have been applied
+      // this means that we need to invert the list to correctly
+      // decode the message
+      encodings.reverse();
+
+      if (encodings.length > this.maxEncodingLayers) {
+        throw new BadRequestError('Too many encodings');
+      }
+
+      for (const encoding of encodings) {
+        let pipe = null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "pipe" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1854)